### ISSUE TITLE
Tileset: Make physics layers nameable

### ISSUE THREAD

**Issue by Proggle** [godotengine/godot-proposals#7720]
### Describe the project you are working on

A 2d game using tilesets.

### Describe the problem or limitation you are having in your project

Currently, physics layers are named based on their index, so you always have 'physics layer 0','physics layer 1', 'physics layer 2' in your tileset.  This is pretty unhelpful when mapping out tiles, since you have to memorize magic numbers when painting physics onto tilesets

In the physics layer list in the tileset itself, the lack of name or label leads to this confusing UI
![image](https://github.com/godotengine/godot-proposals/assets/89282845/9cc03ee4-fb0a-498f-96f2-8cfa078bdccb)

If you want to check the collision bits on physics layer 5, you have to count individual unlabeled entries while scrolling - and since the mouse wheel doesn't smooth scroll on this window, and the layer/mask boxes are set up identically, that's easy to screw up.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

It should be possible to give a defined name to a physics layer, just as with collision layers.  So you could label one of them 'player collision', one of them 'monster collision', et cetera.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Basically, this would involve adding a textbox to a physics layer for the dev to give it a name.  Put it next to where the collision layers/ collision masks are defined.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

This is for editor UI so can't be worked around with script.

### Is there a reason why this should be core and not an add-on in the asset library?

I think this would make the 2D tileset usage a lot easier for developers.

**Comment by kleonc** [comment]
If TileSet's physics layers were to become nameable then it would probably make sense to do the same for other TileSet layers. Custom layers are already nameable, but navigation and occlusion layers are not.
![8MC1Iqhz3c](https://github.com/godotengine/godot-proposals/assets/9283098/1b546560-a50e-4569-90c9-bc8193612854)
And likely the same for terrain sets.
![Godot_v4 1 1-stable_win64_6Yfae5iTWr](https://github.com/godotengine/godot-proposals/assets/9283098/90c12232-77be-46d6-abfb-b524c06ee480)

**Comment by Proggle** [comment]
> If TileSet's physics layers were to become nameable then it would probably make sense to do the same for other TileSet layers. Custom layers are already nameable, but navigation and occlusion layers are not. ![8MC1Iqhz3c](https://user-images.githubusercontent.com/9283098/268749752-1b546560-a50e-4569-90c9-bc8193612854.png) And likely the same for terrain sets. ![Godot_v4 1 1-stable_win64_6Yfae5iTWr](https://user-images.githubusercontent.com/9283098/268749887-90c12232-77be-46d6-abfb-b524c06ee480.png)

Yep, makes sense there too.  And of course just plain tileset 'layers' are already nameable as well, so really we have this feature in 2/5 layer types.
